pysersic/priors.py

```diff
@@ -507,27 +514,33 @@ def generate_sersic_prior(image: jax.numpy.array,
 
     """
     prior = PySersicSourcePrior('sersic', sky_type= sky_type, suffix=suffix)
+    if mask is not None:
+        cat = data_properties(image,mask=mask.astype(bool))
+    else:
+        cat = data_properties(image)
+    
 
     if flux_guess is None:
-        flux_guess = jnp.sum(image)
-    prior.set_gaussian_prior('flux',flux_guess,jnp.sqrt(flux_guess))
+        flux_guess = cat.segment_flux
+    prior.set_gaussian_prior('flux',flux_guess,2*jnp.sqrt(flux_guess))
     
 
-    image_dim_min = jnp.min(jnp.array([image.shape[0],image.shape[1]])) 
+    
     if r_eff_guess is None:
-        r_eff_guess = image_dim_min/10
+        r_eff_guess = (cat.semimajor_sigma/2).value
     
     r_loc = r_eff_guess
     r_scale = jnp.sqrt(r_eff_guess) 
     prior.set_truncated_gaussian_prior('r_eff', r_loc,r_scale, low = 0.5)
 
     prior.set_uniform_prior('ellip', 0,0.9)
-    prior.set_custom_prior('theta', dist.VonMises(loc = 0,concentration=0), reparam= infer.reparam.CircularReparam() )
+    theta_guess = cat.orientation.to(u.rad).value
+    prior.set_custom_prior('theta', dist.VonMises(loc = theta_guess,concentration=0), reparam= infer.reparam.CircularReparam() )
```

Comment:
I concentration of 0 implies flat distribution I think, see : https://en.wikipedia.org/wiki/Von_Mises_distribution, maybe set to 2 or something?